### PR TITLE
Improvements to wandb restore to handle the wrong git repo

### PR DESCRIPTION
**Pull request by vanpelt**

### CODE REVIEW

File: wandb/meta.py [hunk 1]
```diff
@@ -72,6 +72,10 @@ def setup(self):
             self.data["program"] = __main__.__file__
         except (ImportError, AttributeError):
             self.data["program"] = '<python with no main file>'
+        # TODO: we should use the cuda library to collect this
+        if os.path.exists("/usr/local/cuda/version.txt"):
```
Comment (raubitsj) [inline wandb/meta.py]: this path doesnt work on our local gpu (wbeast/wbone) setups:
```
jeff@wbeast:~/work/github/codenet$ ls /usr/local/cuda-10.0/version.txt 
/usr/local/cuda-10.0/version.txt
jeff@wbeast:~/work/github/codenet$ ls -laF /usr/local/
total 52
drwxr-xr-x 13 root root 4096 Dec 15 19:50 ./
drwxr-xr-x 13 root root 4096 Dec 16 12:38 ../
drwxr-xr-x  2 root root 4096 Dec 16 13:15 bin/
drwxr-xr-x 16 root root 4096 Sep 24 16:08 cuda-10.0/
drwxr-xr-x  3 root root 4096 Dec 15 20:03 cuda-8.0/
drwxr-xr-x  4 root root 4096 Feb 13  2018 cuda-9.0/
drwxr-xr-x  2 root root 4096 Aug  1  2017 etc/
drwxr-xr-x  2 root root 4096 Aug  1  2017 games/
drwxr-xr-x  2 root root 4096 Aug  1  2017 include/
drwxr-xr-x  4 root root 4096 Aug  9  2017 lib/
lrwxrwxrwx  1 root root    9 Aug 10  2017 man -> share/man/
drwxr-xr-x  2 root root 4096 Aug  9  2017 sbin/
drwxr-xr-x 10 root root 4096 Aug 14  2017 share/
drwxr-xr-x  2 root root 4096 Aug  1  2017 src/
```
I am not sure what the standard setup is, if people often have multiple cuda versions installed at once like we do.

File: wandb/cli.py [hunk 2]
```diff
@@ -313,15 +311,22 @@ def restore(ctx, run, branch, project, entity):
         project, run = rest.split(":", 1)
```
[the remark this answers is not in the document]
Comment (vanpelt) [reply wandb/cli.py]: I *believe the client currently strips all non-word characters from run_id...  I'll double check.

File: wandb/meta.py (hunk 1 above)
Comment (vanpelt) [reply wandb/meta.py]: Yeah, there's a better way to do it, hence the comment, but figured having something is better than nothing.

File: wandb/cli.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (raubitsj) [reply wandb/cli.py]: FYI: https://app.wandb.ai/jeffr/region-obstruction/runs/i:have:colons

**Pull request merged**